tests/search/grouping_adv/grouping_base.rb

```diff
@@ -94,6 +94,24 @@ def querytest_common
     check_query('all(group(time.dayofweek(from)) each(output(count()) ))',
                 'time.wday')
 
+    # Test time with timezone
+    check_query('all(group(time.year(from)) each(output(count()) ))',
+                'time.year', DEFAULT_TIMEOUT, true, "default", "America/Los_Angeles")
+    check_query('all(group(time.monthofyear(from)) each(output(count()) ))',
+                'time.month', DEFAULT_TIMEOUT, true, "default", "America/Los_Angeles")
+    check_query('all(group(time.hourofday(from)) each(output(count()) ))',
+                'time.hour.timezone', DEFAULT_TIMEOUT, true, "default", "America/Los_Angeles")
+    check_query('all(group(time.secondofminute(from)) each(output(count()) ))',
+                'time.second', DEFAULT_TIMEOUT, true, "default", "America/Los_Angeles")
+    check_query('all(group(time.minuteofhour(from)) each(output(count()) ))',
+                'time.minute.timezone', DEFAULT_TIMEOUT, true, "default", "Australia/Adelaide")
+    check_query('all(group(time.dayofmonth(from)) each(output(count()) ))',
+                'time.mday.timezone', DEFAULT_TIMEOUT, true, "default", "Pacific/Kiritimati")
+    check_query('all(group(time.dayofyear(from)) each(output(count()) ))',
+                'time.yday.timezone', DEFAULT_TIMEOUT, true, "default", "America/Los_Angeles")
+    check_query('all(group(time.dayofweek(from)) each(output(count()) ))',
+                'time.wday.timezone', DEFAULT_TIMEOUT, true, "default", "Europe/Oslo")
```

Comment:
A brief comment on why these particular time zone identifiers were chosen would be nice, especially for some of the more... _esoteric_ ones 🙂

Also would be nice with a comment on what setting the timezone actually implies.